specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+* Payment clients
+* DRM clients and content protection
+* Secure storage
+* User identity management
+* Attestation services
+
+The TEE model divides software into physically isolated domains:
+
+* Normal domain +
+Typically hosting a _rich OS_ (for example, RTOS or Linux), and user applications. 
+* TEE domain +
+Hosts a _TEE OS_ (domain security manager) and _trusted applications (TA)_. 
+* Root domain +
+Hosts RoT firmware, including a secure monitor.
+
+The TEE OS is primarily responsible for isolation of TA, and for providing root of trust services, within the TEE domain.
+
+The OS in Normal domain typically controls scheduling on the system, across all Harts available to it. To interact with TA services in TEE domain, the OS in Normal domain interacts with a TEE OS through a secure monitor in Root domain. 
+
+The secure monitor is responsible for context switching and isolation across domain boundaries, including event management. 
+
+For the purpose of this specification, TEE deployment models can be separated as:
+
+* Static partition TEE +
+A single TEE provides security services to Normal domain. TA are typically installed at boot by RoT FW and TEE OS, though Global Platform does also define protocols for installation of TA at runtime. System configuration and resource allocation can be mostly static, making the system more deterministic. +
+ +
+_Use case examples:_ edge devices and IoT, automation, and automotive. 
+* Virtualized TEE +
+On a virtualized system, TEE can also be virtualized. In this case a _secure partition manager_ in TEE domain is responsible for isolation of multiple TEE guests (for example, an OEM TEE and separate third party TEE). This model can also support more dynamic resource allocation. +
+ +
+_Use case examples:_ mobile clients, and automotive.
+
 ==== Isolation model
 
-==== Device access control
+A Global Platform TEE requires the following isolation guarantees:
 
-==== Sealing
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Root domain MAY access resources assigned to any domain, but SHOULD prevent itself from unintended access to resources assigned to a different domain (privilege escalation).
+
+| CAT_NNN
+| No other domains can access resources assigned to Root domain
+
+| CAT_NNN
+| Resources assigned to TEE domain MUST NOT be accessible to Normal domain
+
+| CAT_NNN
+| Resources assigned to Normal domain MUST be accessible to Normal domain (r/w/x), and to TEE domain (r/w) (default sharing rule)
+
+| CAT_NNN
+| Resources assigned to a single TA, or a guest TEE, MUST not be accessible by a different TA, or guest TEE.
+
+|===
+
+The standard GP TEE model does not support sharing memory between TA in TEE domain. Each TA is expected to be a self-contained unit providing a specific security service, either to Normal domain or to other TA. All communications are implemented through secure channels managed by the TEE OS or SPM. 
```

Comment:
GP model doesnt support this, but i have seen systems requiring inter TA communication and shared memory. for instance watermarking  and secure video pipe management.